pkg/controller/windowsmachine/metrics/metrics.go

```diff
@@ -84,19 +85,36 @@ func Add(ctx context.Context, cfg *rest.Config, namespace string) error {
 		return errors.Wrap(err, "could not create metrics Service")
 	}
 
-	// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
-	// necessary to configure Prometheus to scrape metrics from this operator.
-	services := []*v1.Service{service}
-	_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
+	// Create a monitoring client
+	mclient, err := monclient.NewForConfig(cfg)
 	if err != nil {
-		log.Error(err, "could not create ServiceMonitor object")
-		// If this operator is deployed to a cluster without the prometheus-operator running, it will return
-		// ErrServiceMonitorNotPresent, which can be used to safely skip ServiceMonitor creation.
-		if err == metrics.ErrServiceMonitorNotPresent {
-			metricsEnabled = false
-			return errors.Wrap(err, "install prometheus-operator in your cluster to create ServiceMonitor objects")
+		return errors.Wrap(err, "could not create monitoring client")
+	}
 
+	// Check for existing SM object. If one exists, delete. Else create.
+	existingSM, err := mclient.ServiceMonitors(namespace).Get(context.TODO(), windowsMetricsEndpoints, metav1.GetOptions{})
+	if existingSM != nil {
+		deleteErr := mclient.ServiceMonitors(namespace).Delete(context.TODO(), windowsMetricsEndpoints, metav1.DeleteOptions{})
+		if deleteErr != nil {
+			return errors.Wrap(err, "could not delete existing ServiceMonitor object")
+		}
+	} else {
+		if apierrors.IsNotFound(err) {
+			services := []*v1.Service{service}
+			// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
+			// necessary to configure Prometheus to scrape metrics from this operator.
+			_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
+			if err != nil {
+				log.Error(err, "could not create ServiceMonitor object")
+				// If this operator is deployed to a cluster without the prometheus-operator running, it will return
+				// ErrServiceMonitorNotPresent, which can be used to safely skip ServiceMonitor creation.
+				if err == metrics.ErrServiceMonitorNotPresent {
+					metricsEnabled = false
+					return errors.Wrap(err, "install prometheus-operator in your cluster to create ServiceMonitor objects")
+				}
+			}
 		}
+		return errors.Wrap(err, "could not get ServiceMonitor object")
```

Comment:
```suggestion
	}
	// The ServiceMonitor object was either not found or was deleted
	if (err != nil && apierrors.IsNotFound(err)) || existingSM != nil {
		services := []*v1.Service{service}
		// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
		// necessary to configure Prometheus to scrape metrics from this operator.
		_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
		if err != nil {
			log.Error(err, "could not create ServiceMonitor object")
			// If this operator is deployed to a cluster without the prometheus-operator running, it will return
			// ErrServiceMonitorNotPresent, which can be used to safely skip ServiceMonitor creation.
			if err == metrics.ErrServiceMonitorNotPresent {
				metricsEnabled = false
				return errors.Wrap(err, "install prometheus-operator in your cluster to create ServiceMonitor objects")
			}
			return errors.Wrap(err, "could not create ServiceMonitor object")
		}
	}
	if err != nil && !apierrors.IsNotFound(err) {
		return errors.Wrap(err, "could not get ServiceMonitor object")
	}
```
An completely untested suggestion that might fix my other comments :-) WDYT?

Reply:
Thanks Aravindh! I'll test these changes. In line 103, I think  existingSM would need to be `existingSM == nil` because in the case that existingSM was deleted it would be empty. Also do we need `could not create ServiceMonitor object`error message twice (lines 109 and 116)? Or is it fine just to have it on line 116?

Reply:
Why would the variable `existingSM` be `nil` because the object was deleted? The variable `existingSM` will still hold the object data. It is just that it will be stale. 

I am fine not having the log (line 109) and just reflecting it when returning the error (line 116).

Reply:
I found my previous approach to be faulty. Please see my latest commit.

Reply:
There was an error where existingSM != nil was always true and a SM was not able to be created.